src/Illuminate/Database/Eloquent/Collection.php

```
@@ -843,4 +845,44 @@ public function toQuery()
 
         return $model->newModelQuery()->whereKey($this->modelKeys());
     }
+
+    /**
+     * Create a new resource collection instance for the given resource.
+     *
+     * @param  class-string<JsonResource>|null  $resourceClass
+     * @return ResourceCollection
+     */
+    public function toResourceCollection(?string $resourceClass = null): ResourceCollection
+    {
+        if ($resourceClass === null) {
+            return $this->guessResourceCollection();
+        }
+
+        return $resourceClass::collection($this);
+    }
+
+    /**
+     * Guess the resource collection for the items.
+     *
+     * @return ResourceCollection
+     */
+    protected function guessResourceCollection(): ResourceCollection
+    {
+        if ($this->isEmpty()) {
+            return new ResourceCollection($this);
+        }
+
+        $model = $this->first();
+
+        assert(is_object($model), 'Resource collection guesser expects the collection to contain objects.');
```

Using assertions in production code is risky as they may be disabled in production environments (zend.assertions=-1). When assertions fail in this state, no proper errors are thrown, leading to silent failures or unexpected behavior. Better to use explicit exceptions with meaningful error messages for reliable error handling.

https://www.php.net/manual/en/ini.core.php#ini.zend.assertions